feat: support proxy and mTLS in import.git (#4313)
grafana/alloy: docs/sources/reference/config-blocks/import.git.md

| -------------------------- | ------------------------------------------------------------ | -------- |
| [`basic_auth`][basic_auth] | Configure `basic_auth` for authenticating to the repository. | no       |
| [`ssh_key`][ssh_key]       | Configure an SSH Key for authenticating to the repository.   | no       |

> | [`tls_config`][tls_config] | Configure TLS settings for connecting to the endpoint.       | no       |

Reviewer: We are missing the cross ref link for `tls_config`. The other links are defined on lines 142-144. The link for `tls_config` could be added on line 145... or under the table. Either place is OK.

The link definition will look like this:

`[tls_config]: #tls_config`

| ---------------------- | -------- | -------------------------------------------------------- | ------- | -------- |
| `ca_pem`               | `string` | CA PEM-encoded text to validate the server with.         |         | no       |
| `ca_file`              | `string` | CA certificate to validate the server with.              |         | no       |
| `cert_pem`             | `string` | Certificate PEM-encoded text for client authentication.  |         | no       |
| `cert_file`            | `string` | Certificate file for client authentication.              |         | no       |
| `insecure_skip_verify` | `bool`   | Disables validation of the server certificate.           |         | no       |
| `key_file`             | `string` | Key file for client authentication.                      |         | no       |
| `key_pem`              | `secret` | Key PEM-encoded text for client authentication.          |         | no       |

The following pairs of arguments are mutually exclusive and can't both be set simultaneously:

* `ca_pem` and `ca_file`
* `cert_pem` and `cert_file`
* `key_pem` and `key_file`

> When configuring client authentication, both the client certificate (using `cert_pem` or `cert_file`) and the client key (using `key_pem` or `key_file`) must be provided.

Reviewer: ```suggestion
When you configure client authentication, you must provide both the client certificate with`cert_pem` or `cert_file`, and the client key with `key_pem` or `key_file`.
```

Making a little more direct and active and removing the brackets.

Author: Thanks for the review @clayton-cornell ! I've copied this part from `docs/sources/shared/reference/components/tls-config-block.md` without `min_version` and `server_name` (because go-git doesn't support it). Do you want me to also make the change in the original file?

Reviewer: No it's OK. We can catch that shared topic another time.
